### PR TITLE
John conroy/try GitHub actions

### PR DESCRIPTION
**Pull request by john-conroy**
A work in progress, but I think this would be the general idea. We may have to modify `test.sh` and `portal-ci.yml` to make this work. Right now I'm just making `test.sh` flexible enough to run both github actions and travis.

I was getting the error below and had to remove the `tput` code from `docker.sh`.
```
tput: No value for $TERM and no -T specified
Error: Process completed with exit code 2.
```

### CODE REVIEW

File: .github/workflows/portal-ci.yml
Comment (mccalluc) [inline .github/workflows/portal-ci.yml:24]: I have this memory of reading that caching pip in CI doesn't help much... There isn't that much repeated computation that's being saved... but maybe the github cache is significantly closer or faster than pypi?

File: .github/workflows/portal-ci.yml
Comment (mccalluc) [inline .github/workflows/portal-ci.yml:32]: Huh... `submodules: 'recursive'` above isn't sufficient?

File: docker.sh
Comment (mccalluc) [inline docker.sh:28]: Consider 
```
green=`tput setaf 2 || echo ''`
reset=`tput sgr0 || echo ''`
```
... though it'd be nice to understand why it won't work, but don't let that block the PR.

**Review by mccalluc — COMMENTED** [review]
Nifty. Before merging, would like to understand the recursive checkout, and keep the tput during local development.

Not the highest priority, but is there an easy way to do output folding on github?

File: .github/workflows/portal-ci.yml
Comment (john-conroy) [reply .github/workflows/portal-ci.yml:32]: Isn't it necessary since `dev-start` has the below?

```
[ "`git config --get submodule.recurse`" == 'true' ] \
  || die "To update submodules automatically: git config --global submodule.recurse true";
```

File: docker.sh
Comment (john-conroy) [reply docker.sh:28]: Thanks, I'll add it back in.

File: .github/workflows/portal-ci.yml
Comment (john-conroy) [reply .github/workflows/portal-ci.yml:24]: I just did this to mirror what we have in travis, I can remove it for simplicity?

Comment (mccalluc) [reply .github/workflows/portal-ci.yml:24]: Not a strong preference... if it's in, and it works, fine to leave it, I think.

File: .github/workflows/portal-ci.yml
Comment (mccalluc) [reply .github/workflows/portal-ci.yml:32]: Got it. I think you're right: What you're doing is the simplest way to do this.

File: .github/workflows/portal-ci.yml
Comment (mccalluc) [inline .github/workflows/portal-ci.yml:31]:
```suggestion
    - name: Set up git config to make dev-start.sh happy
```

**Review by mccalluc — APPROVED** [review]
Looks good... One final oddity: Travis isn't running, but I don't think it's been disabled? I guess if Github CI is turned on, it doesn't notify Travis? In that case, yes, let's drop the travis.yml... We can always get it from the history, if we need it.

**Pull request merged**
